crates/chain/src/tx_data_traits.rs

```diff
@@ -31,3 +34,50 @@ impl ForEachTxOut for Transaction {
         }
     }
 }
+
+/// Trait that "anchors" blockchain data to a specific block of height and hash.
+///
+/// I.e. If transaction A is anchored in block B, then if block B is in the best chain, we can
+/// assume that transaction A is also confirmed in the best chain. This does not necessarily mean
+/// that transaction A is confirmed in block B. It could also mean transaction A is confirmed in a
+/// parent block of B.
```

Comment:
Looking at `Anchor`'s definition, I'm wondering if there's something else it could make sense to implement it on, other than `BlockId`. 
I'm not saying that Anchor shouldn't be a generic, I'm just wondering if you thought of any other structure that could be used as a anchor :)

Reply:
`Anchor` implementations can be a lot more expressive. For example, `Anchor` with the exact confirmation height and/or time of the transaction. Or an `Anchor` with the exact merkle proof.

Reply:
FWIW I think anchor should not be implemented on `BlockId`. The semantics of it are ambiguous. Does it imply that the transaction is anchored to that block because the transaction is in the block or does that block merely imply that the transaction is in the block? I didn't mention this because I thought the discussion would come up later anyway.